pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -224,66 +222,122 @@ func logMetricStatusToKRM(in *api.LogMetric, out *krm.LoggingLogMetricStatus) er
 }
 
 func (a *logMetricAdapter) Update(ctx context.Context, u *unstructured.Unstructured) error {
-	update := new(api.LogMetric)
-	*update = *a.actual
-
-	if ValueOf(a.desired.Spec.Description) != a.actual.Description {
-		update.Description = ValueOf(a.desired.Spec.Description)
-	}
-	if ValueOf(a.desired.Spec.Disabled) != a.actual.Disabled {
-		update.Disabled = ValueOf(a.desired.Spec.Disabled)
-	}
-	if a.desired.Spec.Filter != a.actual.Filter {
-		// todo acpana: revisit UX, err out if filter of desired is empty
-		if a.desired.Spec.Filter != "" {
-			update.Filter = a.desired.Spec.Filter
-		} else {
-			// filter is a REQUIRED field
-			update.Filter = a.actual.Filter
+	latest := a.actual
+
+	if a.hasChanges(ctx, u) {
+		update := new(api.LogMetric)
+		*update = *a.actual
+
+		if ValueOf(a.desired.Spec.Description) != a.actual.Description {
+			update.Description = ValueOf(a.desired.Spec.Description)
+		}
+		if ValueOf(a.desired.Spec.Disabled) != a.actual.Disabled {
+			update.Disabled = ValueOf(a.desired.Spec.Disabled)
+		}
+		if a.desired.Spec.Filter != a.actual.Filter {
+			// todo acpana: revisit UX, err out if filter of desired is empty
+			if a.desired.Spec.Filter != "" {
+				update.Filter = a.desired.Spec.Filter
+			} else {
+				// filter is a REQUIRED field
+				update.Filter = a.actual.Filter
```

Comment:
Just FYI, passing empty Filter to logging server is allowed. It will do a no-op.

Reply:
We need a test here IMO

Reply:
> Just FYI, passing empty Filter to logging server is allowed. It will do a no-op.

From a previous channel, while the current dcl controller allows to send an ""/ empty filter to the platform, it errors out  ...